src/test/java/com/conveyal/datatools/manager/jobs/AutoDeployFeedJobTest.java

```diff
@@ -110,26 +126,55 @@ public void canAutoDeployFeedVersion() throws IOException {
         setFeedSourceDeployable(true);
         setProjectAutoDeploy(true);
         setProjectPinnedDeploymentId(deployment.id);
-        ProcessSingleFeedJob processSingleFeedJob = triggerProcessSingleFeedJob("fake-agency-expire-in-2099");
+        ProcessSingleFeedJob processSingleFeedJob = triggerProcessSingleFeedJob(mockFeedSource1, "fake-agency-expire-in-2099");
         MonitorableJob lastJob = getLastJob(processSingleFeedJob);
         assertTrue(lastJob instanceof AutoDeployFeedJob);
         assertThat(lastJob.status.message, equalTo("Job complete!"));
     }
 
+    @Test
+    public void failAutoDeployIfFetchStillInProgress() throws IOException {
+        setFeedSourceDeployable(true);
+        setProjectAutoDeploy(true);
+        setProjectPinnedDeploymentId(deployment.id);
+
+        // Mock fake fetch to satisfy second condition of Deployment.hasFeedFetchesInProgress.
+        File zipFile = zipFolderFiles("fake-agency-expire-in-2099");
+        feedVersion = getFeedVersionFromGTFSFile(mockFeedSource2, zipFile);
+        Auth0UserProfile user = Auth0UserProfile.createTestAdminUser();
+        Set<MonitorableJob> userJobs = Sets.newConcurrentHashSet();
+        userJobs.add(new ProcessSingleFeedJob(feedVersion, user, true));
+        DataManager.userJobsMap.put(user.getUser_id(), userJobs);
+
+        // Add the feed version id to the deployment to satisfy first condition of Deployment.hasFeedFetchesInProgress.
```

Comment:
I found this comment somewhat confusing. Maybe instead it could read:
```
Add mock feed 1 to the deployment so that it is detected in the Deployment#hasFeedFetchesInProgress check (called during mock feed 2's processing).
```

Reply:
Updated.